core/tests/test_utils.py

```diff
@@ -71,11 +72,27 @@ def __init__(self, service_name='urlfetch'):
         self.request = None
 
     def set_return_values(self, content='', status_code=200, headers=None):
+        """Set mock content, status_code and headers for Url Service mock
+        object's return_values.
+
+        Args:
+            content: Content for the return_values. Defaults to empty string.
+            status_code: status_code for the return_values. Defaults to 200.
+            headers: headers for the return_values. Defaults to None.
+
+        """
         self.return_values['content'] = content
         self.return_values['status_code'] = status_code
         self.return_values['headers'] = headers
 
-    def _Dynamic_Fetch(self, request, response): # pylint: disable=invalid-name
+    def _Dynamic_Fetch(self, request, response):  # pylint: disable=invalid-name
+        """
+        Sets request and response content for the URL Mock.
```

Comment:
I'm not sure this is the point of this method, otherwise what's the rationale behind lines 96-103?

What calls _Dynamic_Fetch? I would have assumed that this simulates a url fetch.